tests/events/events_v2.test.js

```diff
@@ -0,0 +1,81 @@
+import { beforeAll, describe, expect, it } from '@jest/globals'
```

Comment:
We should expand these tests to test that the events logic is wired into the engine correctly. I can see here that you check that the listener is created, but it would be good to also check that instances can be created from events and existing instances waiting on events wake up and proceed correctly. This can be split into another ticket to avoid holding up the PR.

Reply:
I created DIR-1511 for that. Also all tests from v1 will be migrated to v2 eventually